pkg/controllers/scheduler/framework/util.go

```diff
@@ -245,6 +250,18 @@ func (r *Resource) SetMaxResource(rl corev1.ResourceList) {
 	}
 }
 
+// HasGivenScalarResource checks if Resource has the given scalar resource.
+func (r *Resource) HasGivenScalarResource(name corev1.ResourceName) bool {
+	if r.ScalarResources == nil {
+		return false
+	}
+
+	if _, exists := r.ScalarResources[name]; exists {
```

Comment:
Why should one use this function instead of indexing `r.ScalarResources` directly?

Reply:
Mainly to avoid the situation where ScalarResources is nil.

Reply:
Reading a nil map does not panic; only writing to a nil map panics